test/e2e/set/scyllacluster/helpers.go

```diff
@@ -74,7 +86,14 @@ func waitForScyllaClusterState(ctx context.Context, client scyllav1client.Scylla
 	event, err := watchtools.UntilWithSync(ctx, lw, &scyllav1.ScyllaCluster{}, nil, func(event watch.Event) (bool, error) {
 		switch event.Type {
 		case watch.Added, watch.Modified:
-			return condition(event.Object.(*scyllav1.ScyllaCluster))
+			for _, condition := range conditions {
+				if ok, err := condition(event.Object.(*scyllav1.ScyllaCluster)); err != nil {
```

Comment:
nit: I'd take out the assignment out of the if condition to reduce the line length and avoided needing else if